app/src/org/commcare/tasks/CleanRawMedia.kt

```diff
@@ -0,0 +1,31 @@
+package org.commcare.tasks
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.commcare.CommCareApplication
+import org.commcare.android.database.user.models.FormRecord
+import org.commcare.utils.CrashUtil
+import org.commcare.utils.StorageUtils.getUnsentCompleteOrSavedFormIdsForCurrentApp
+import java.io.File
+
+class CleanRawMedia(appContext: Context, workerParams: WorkerParameters)
+    : Worker(appContext, workerParams) {
+
+    override fun doWork(): Result {
+        val storage = CommCareApplication.instance().getUserStorage(FormRecord::class.java)
+        val recordsToRemove = getUnsentCompleteOrSavedFormIdsForCurrentApp(storage)
+        for (formId in recordsToRemove) {
+            try {
+                val filePath = storage.getMetaDataFieldForRecord(formId, FormRecord.META_FILE_PATH)
+                File(filePath).parent?.let {
+                    val rawDirPath = "$it/raw"
+                    File(rawDirPath).deleteRecursively()
+                }
+            } catch (e: Exception) {
+                CrashUtil.reportException(e)
```

Comment:
I think we should fail at the bottom of the loop still if one iteration didn't process right? That will let it retry later, although if we are launching at-most-one of these per startup maybe it's not worth worrying about the distinciton.

Reply:
We are launching these at-most-once on a user login, though still think that it's not worth the distinction here as it's not crticial to keep retrying for a few failed records.